aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -114,9 +140,99 @@ macro_rules! unwrap_or_discard {
     };
 }
 
+/// Public interface for the VM.
 impl AptosVM {
+    #[allow(clippy::new_without_default)]
     pub fn new(state: &impl StateView) -> Self {
-        Self(AptosVMImpl::new(state))
+        let storage = StorageAdapter::new(state);
+
+        // Get the gas parameters
```

Comment:
maybe some of this gas things can be extracted out to utils.rs (or gas_utils.rs if there are more things)